### PR TITLE
Rename services types[].type to types[].name; Drop /types endpoint

### PR DESCRIPTION
**Pull request by n3wscott**
Fixes: https://github.com/cloudevents/spec/issues/685
Fixes: https://github.com/cloudevents/spec/issues/684
Fixes: https://github.com/cloudevents/spec/issues/679

- Rename services types[].type to types[].name
- Drop /types endpoint

Signed-off-by: Scott Nichols <snichols@vmware.com>

### CODE REVIEW

File: discovery.md [hunk 1]
```diff
@@ -41,19 +41,8 @@ a Discovery Endpoint could choose to be implemented as part of a Service, or
...
-
-The second case becomes relevant if multiple services support the same event
-types. Use case 1 is likely the dominant use case. Given the example of a public
-cloud provider where all services generate events, there might be dozens of
-sources and hundreds of event types. The discovery funnel of services first,
-then event types helps users navigate without having to see large lists of event
-types. Both of these cases show the importance of using filters in the discovery
-API to narrow down the selection of available events.
+The main usecase to condider from the viewpoint of the event consumer is what
+services are available, and what event types do they generate?
```
Comment (deissnerk) [inline discovery.md:45]: Why do we drop the second use case entirely? I think it is still relevant.

File: discovery.md [hunk 2]
```diff
@@ -156,7 +145,7 @@ Service:
   "protocols": [ "[string]" + ],
   "types": [ ?
     { *
-      "type": "[ce-type value]",
+      "name": "[ce-type value]",
```
Comment (deissnerk) [inline discovery.md:148]: I wonder, if it is the attribute `type` that needs to be renamed to `name`, or if the name `types` of the array is misleading. Is it really a list of types? We are specifying how context attributes (including `type`) are used. So isn't it a list of event pattern or definitions?

Comment (deissnerk) [reply discovery.md:148]: One more thing:
Would `name` be unique within the elements of `types`, or could there be multiple entries for the same event type?

File: discovery.md [hunk 3]
```diff
@@ -41,19 +41,8 @@ a Discovery Endpoint could choose to be implemented as part of a Service, or
...
-2. What event types are available, and from which services?
-
-The second case becomes relevant if multiple services support the same event
-types. Use case 1 is likely the dominant use case. Given the example of a public
-cloud provider where all services generate events, there might be dozens of
-sources and hundreds of event types. The discovery funnel of services first,
-then event types helps users navigate without having to see large lists of event
-types. Both of these cases show the importance of using filters in the discovery
-API to narrow down the selection of available events.
+The main usecase to condider from the viewpoint of the event consumer is what
```
Comment (duglin) [inline discovery.md:44]: s/condider/consider/

File: discovery.md (hunk 2 above)
Comment (duglin) [reply discovery.md:148]: hmmm s/types/events/ might sounds interesting.  If we did that, would we keep "type"?  Seems like it might be ok then, no?

Comment (n3wscott) [reply discovery.md:148]: Yeah events and type seems fine to me.

File: discovery.md (hunk 1 above)
Comment (n3wscott) [reply discovery.md:45]: I dropped it because the answer to the question is suppose to be calls to the /types endpoint, which I dropped.

Comment (duglin) [reply discovery.md:45]: It does beg the question... how does someone get a list of all of the 'types'?  Or is that even a valid question? Should they be able to ask for the list of all "sourcetemplates" too?  With the rename of "types" to "events", is "type" no longer special? Should it be?

I like the idea of treating all attributes equally. So, perhaps we should ask if getting the list of types is really a query people want to do?  Or is it more likely they people have a type in mind (e.g. com.example.db.created) and are looking for services that emit that?  Is a list of random strings (types) something that's interesting and useful?

Comment (duglin) [reply discovery.md:45]: I guess if someone can explain why the list of types is useful then that would help us define the `/types` API that this PR is removing.

**Comment by duglin** [comment]
Approved on the 8/27 call

**Pull request merged**